src/adaptors/curve/index.js

```diff
@@ -1,269 +1,87 @@
-    }`
-  );
-  tvl = tvl.data.pools;
-
-  const poolNames = Object.keys(tvl);
-  let dataTvl = [];
-  for (const p of poolNames) {
-    const obj = tvl[p];
-    obj['pool'] = p;
-    dataTvl.push(obj);
+const getCurvePoolsAPY = async (blockchainId) => {
+  const uri = `/getSubgraphData/${blockchainId}`;
+  let response;
+  try {
+    response = await utils.getData(CRV_API_BASE_URL + uri);
+  } catch (error) {
+    return {};
   }
-
-  if (chainString === 'fantom') {
-    dataApy = await utils.getData(
-      `https://stats.curve.fi/raw-stats-ftm/apys.json`
-    );
-  } else {
-    dataApy = await utils.getData(
-      `https://stats.curve.fi/raw-stats-${chainString}/apys.json`
-    );
-  }
-  for (const el of dataTvl) {
-    el.apy = dataApy.apy.day[el.pool];
+  const poolAPY = {};
+  if (response?.success) {
+    for (const pool of response.data.poolList) {
+      poolAPY[pool.address] = { apy: pool.latestDailyApy * 100 };
+    }
   }
-
-  dataTvl = dataTvl.filter((el) =>
-    Object.keys(pools.tokenMapping[chainString]).includes(el.pool)
-  );
-
-  return dataTvl;
-};
-
-const buildPool = (entry, chainString) => {
-  const token = pools.tokenMapping[chainString][entry.pool];
-  const id = pools.pools[chainString].find(
-    (el) => el.name === entry.pool
-  )?.swap;
-  let apy = entry.apy < 0 ? 0 : entry.apy;
-  apy = chainString === 'ethereum' ? apy : apy * 100;
-
-  const newObj = {
-    // ugly, but some swap pools have same address on diff networks
-    pool: `${id}-${chainString}`,
-    chain: utils.formatChain(chainString),
-    project: 'curve',
-    symbol: utils.formatSymbol(token),
-    tvlUsd: entry.tvl,
-    apy: apy,
-  };
-  return newObj;
+  return poolAPY;
 };
 
-const topLvl = async (chainString) => {
-  if (chainString === 'ethereum') {
-    // pull data
-    poolStats = await getDataEth();
-  } else {
-    // pull data
-    poolStats = await getDataEVM(chainString);
-  }
-  // build pool objects
-  let data = poolStats.map((el) => {
-    if (pools.tokenMapping[chainString][el.pool] === undefined) {
-      return null;
-    }
-    return buildPool(el, chainString);
+const main = async () => {
+  const defillamaPooldata = [];
+
+  results = BLOCKCHAINIDS.map((blockchainId) => {
+    return Promise.all([
+      getCurvePoolsInfo(blockchainId),
+      getCurvePoolsAPY(blockchainId),
+    ]).then((poolsInfoAndAPY) => {
+      const [addressToPoolInfo, addressToPoolAPY] = poolsInfoAndAPY;
+      for (const [address, poolInfo] of Object.entries(addressToPoolInfo)) {
+        const pool = Object.assign(poolInfo, addressToPoolAPY[address]);
+
+        if (pool.apy && pool.tvlUsd) {
+          defillamaPooldata.push({
+            pool: address + '-' + blockchainId,
+            chain: blockchainId[0].toUpperCase() + blockchainId.slice(1),
+            project: 'curve',
+            symbol: pool.symbol,
+            tvlUsd: pool.tvlUsd ? pool.tvlUsd : 0,
+            apy: pool.apy ? pool.apy : 0,
```

Comment:
Also we added new fields (check https://github.com/DefiLlama/yield-server/pull/73) so u can split different apy type (feeApy = base curve apy, rewardApy = apy from reward tokens)

Reply:
since crv rewards can be boosted depending on the veCRV, there's a range possible. I'd say the min value is the one that should be considered. What do you think?

Reply:
Yes, I think it's okay to just include the min guaranteed as rewardApy